### PR TITLE
feat: seller payouts

### PR DESCRIPTION
**Pull request by JustSamuel**
# Description
If seller-payout was so good why is there not a seller-payout-2?

Adds a general seller dropdown in the following form:
image

From this you can navigate to a user you are the admin of
image

And the BACPM is able to create a payout:
image

The flow is as follows, the BACPM enters a start and end date:
image

The BACPM will be prompted to `Calculate Amount`, in practice this simply generates a seller report for the given time period:
image

This will, and should be, the amount on the seller payout. The BACPM is also able to (and should!) download the corresponding pdf report. Note that this is not the PDF to put in the bookkeeping, but merely a sanity check.

This also adds the ability to `verify` payouts:
image

Verify means to check if the amount on the payout is still equal to the payout. If not, history has been changed.

## Types of changes
- New feature _(non-breaking change which adds functionality)_

### CODE REVIEW

File: apps/dashboard/src/components/ActionButton.vue [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  },
+});
+
+const emits = defineEmits(['click']);
+
+const buttonIcon = ref('pi pi-check');
+const buttonSeverity = ref('primary');
+
+const updateResult = () => {
+  console.error('result', props.result);
```
Comment (CodeNamedRobin) [inline apps/dashboard/src/components/ActionButton.vue]: Beep boop

File: apps/dashboard/src/components/CardComponent.vue [hunk 2]
```diff
@@ -42,9 +46,11 @@ const props = defineProps({
 
 const router = useRouter();
 const handleClick = () => {
+  console.error('clicked');
```
Comment (CodeNamedRobin) [inline apps/dashboard/src/components/CardComponent.vue]: beep boop

File: apps/dashboard/src/components/CardComponent.vue [hunk 3]
```diff
@@ -42,9 +46,11 @@ const props = defineProps({
...
   if (props.routerLink) {
     // If routerLink is defined, use router.push to navigate
-    router.push({ name: props.routerLink });
+    console.error(props.routerParams);
```
Comment (CodeNamedRobin) [inline apps/dashboard/src/components/CardComponent.vue]: beep boop

File: apps/dashboard/src/components/TopNavbar.vue [hunk 4]
```diff
@@ -113,6 +115,47 @@ const isBACPM = () => {
...
 
+const organs: Ref<any[]> = ref([]);
+
+const getOrgans = async () => {
+  organs.value = [];
+  const promises = [];
+  for (let organ of authStore.organs) {
+    promises.push(apiService.balance.getBalanceId(organ.id).then((res) => {
+      organs.value.push({
+        label: organ.firstName + ' ' + organ.lastName,
```
Comment (CodeNamedRobin) [inline apps/dashboard/src/components/TopNavbar.vue]: prefer formatted string with `${organ.firstName} ${organ.lastName}`

File: apps/dashboard/src/components/TopNavbar.vue [hunk 5]
```diff
@@ -113,6 +115,47 @@ const isBACPM = () => {
...
+        route: '/user/' + organ.id,
+        notifications: res.data.amount.amount > 0 ? ' ' : '',
```
Comment (CodeNamedRobin) [inline apps/dashboard/src/components/TopNavbar.vue:128]: ?????

File: apps/dashboard/src/components/TopNavbar.vue [hunk 6]
```diff
@@ -113,6 +115,47 @@ const isBACPM = () => {
...
+      });
+    }));
+  }
+  await Promise.all(promises);
+  organs.value = organs.value.sort((a, b) => a.label.localeCompare(b.label));
+};
+
+const getAllOrgans = async () => {
```
Comment (CodeNamedRobin) [inline apps/dashboard/src/components/TopNavbar.vue:136]: Possible code duplication with `getOrgans` ?

File: apps/dashboard/src/locales/en/common/common.json [hunk 7]
```diff
@@ -7,6 +7,8 @@
         "actions": "Actions",
         "gewisId": "GEWIS ID",
         "downloadPdf": "Download PDF",
+        "pdf": "PDF",
+        "till": "till",
```
Comment (CodeNamedRobin) [inline apps/dashboard/src/locales/en/common/common.json:11]: please remove the `till` key from other more specific translation modules if you want it in common.

File: apps/dashboard/src/locales/en/modules/seller.json [hunk 8]
```diff
@@ -57,6 +57,36 @@
           "selectCategory": "Select category",
           "selectVat": "Select VAT"
         }
+      },
+      "payouts": {
```
Comment (CodeNamedRobin) [inline apps/dashboard/src/locales/en/modules/seller.json:61]: These do not go into the sellers module, but into the financial module.

File: apps/dashboard/src/main.ts [hunk 9]
```diff
@@ -81,7 +81,6 @@ app.component('ToggleButton', ToggleButton);
 app.component('Steps', Steps);
 app.component('Calendar', Calendar);
 app.component('ConfirmDialog', ConfirmDialog);
-app.component('InputSwitch', InputSwitch);
```
Comment (CodeNamedRobin) [inline apps/dashboard/src/main.ts:84]: Why delete this? We still use input switches for boolean form inputs

File: apps/dashboard/src/mixins/verifyPayoutMixin.ts [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+    const verifying = ref<boolean>(false);
+    const verifySuccess = ref<boolean | null>(null);
+    const toast = useToast();
+    const { t } = useI18n();
+
+    const verifyPayout = async (payout: any) => {
+      console.error(payout);
+      if (!payout) return;
+
+      console.error("what");
```
Comment (CodeNamedRobin) [inline apps/dashboard/src/mixins/verifyPayoutMixin.ts]: ah yes

File: apps/dashboard/src/mixins/verifyPayoutMixin.ts [hunk 11]
```diff
@@ -0,0 +1,100 @@
...
+        endDate.setUTCHours(23, 59, 59, 999);
+
+        console.error(payout.requestedBy.id);
+        const result = await ApiService.user.getUsersSalesReport(
+          payout.requestedBy.id,
+          startDate.toISOString(),
+          endDate.toISOString()
+        );
+
+        console.error(result.data);
```
Comment (CodeNamedRobin) [inline apps/dashboard/src/mixins/verifyPayoutMixin.ts]: beep boop

File: apps/dashboard/src/modules/admin/components/users/AdminUserBalance.vue [hunk 12]
```diff
@@ -1,8 +1,15 @@
 <template>
+<!--  :header="$t('c_currentBalance.balance')"-->
```
Comment (CodeNamedRobin) [inline apps/dashboard/src/modules/admin/components/users/AdminUserBalance.vue]: Commented code

File: apps/dashboard/src/modules/admin/components/users/AdminUserBalance.vue [hunk 13]
```diff
@@ -1,8 +1,15 @@
...
+<!--  action="Create payout"-->
+<!--  :routerLink="getRouterLink"-->
+<!--  :routerParams="getRouterParams"-->
+<!--  class="w-19rem"-->
   <CardComponent
-      :header="t('modules.admin.singleUser.balance.header')"
-      :func="userBalance?.fine ? waiveFines : undefined"
-      :action="userBalance?.fine ? t('modules.admin.singleUser.balance.waiveFines') : undefined"
+    :header="t('modules.admin.singleUser.balance.header')"
+    action="Create payout"
```
Comment (CodeNamedRobin) [inline apps/dashboard/src/modules/admin/components/users/AdminUserBalance.vue]: Hardcoded text

File: apps/dashboard/src/modules/admin/views/AdminSingleUserView.vue [hunk 14]
```diff
@@ -53,6 +55,16 @@ onBeforeMount(async () => {
     await router.replace({ path: '/error' });
     return;
   }
+  console.error(mutations.value);
```
Comment (CodeNamedRobin) [inline apps/dashboard/src/modules/admin/views/AdminSingleUserView.vue]: Beep boop

File: apps/dashboard/src/modules/seller/components/seller/SellerPayoutInfo.vue
Comment (CodeNamedRobin) [inline apps/dashboard/src/modules/seller/components/seller/SellerPayoutInfo.vue:1]: Text formatting could use a little help. I would use formatted strings.

File: apps/dashboard/src/modules/seller/components/seller/SellerPayoutInfo.vue [hunk 15]
```diff
@@ -0,0 +1,109 @@
...
+        </span>
+        <span><strong>{{ t('modules.seller.payouts.payout.amount') +':' }}</strong> {{ formatPrice(payout?.amount) }}</span>
+        <span><strong>{{ t('modules.seller.payouts.payout.description') +':' }}</strong> {{ payout?.reference }}</span>
+      </div>
+      <div class="flex flex-row gap-2 justify-content-between w-full mt-3">
+        <div>
+          <Button
+              type="button"
+              icon="pi pi-trash"
+              :label="$t('common.delete')"
```
Comment (CodeNamedRobin) [inline apps/dashboard/src/modules/seller/components/seller/SellerPayoutInfo.vue]: Please don't use global translator function `$t`

File: apps/dashboard/src/modules/seller/components/seller/SellerPayoutInfo.vue [hunk 16]
```diff
@@ -0,0 +1,109 @@
...
+    emits('payout:deleted');
+  } else {
+    toast.add({ severity: 'error', summary: t('common.error'),
+      detail: t('modules.seller.payouts.payout.deleteError') });
+  }
+};
+
+const downloadPdf = async (id: number) => {
+  downloadingPdf.value = true;
+  // const result = await payoutStore.fetchPdf(id);
```
Comment (CodeNamedRobin) [inline apps/dashboard/src/modules/seller/components/seller/SellerPayoutInfo.vue]: Commented code

File: apps/dashboard/src/modules/seller/components/seller/SellerPayoutsTable.vue [hunk 17]
```diff
@@ -0,0 +1,177 @@
...
+};
+
+onMounted(async () => {
+  await loadPayoutRequests();
+});
+
+async function loadPayoutRequests(skip = 0) {
+  isLoading.value = true;
+  const response: PaginatedSellerPayoutResponse = await sellerPayoutStore.fetchPayoutsBy(props.seller.id, 12, skip).catch(async (e) => {
+    console.error(e);
```
Comment (CodeNamedRobin) [inline apps/dashboard/src/modules/seller/components/seller/SellerPayoutsTable.vue]: beep boop

File: apps/dashboard/src/modules/seller/components/seller/SellerPayoutsTable.vue [hunk 18]
```diff
@@ -0,0 +1,177 @@
...
+    totalRecords.value = response._pagination.count || 0;
+  }
+  isLoading.value = false;
+}
+
+async function onPage(event: DataTablePageEvent) {
+  await loadPayoutRequests(event.first);
+}
+
+// watch(() => payoutStore.getUpdatedAt, () => {
```
Comment (CodeNamedRobin) [inline apps/dashboard/src/modules/seller/components/seller/SellerPayoutsTable.vue]: Commented code

File: apps/dashboard/src/modules/seller/components/seller/forms/SellerPayoutCreateForm.vue [hunk 19]
```diff
@@ -0,0 +1,167 @@
...
+  payoutAmount.value = null;
+});
+
+const updateDate = (field: string, value: string) => {
+  props.form.context.setFieldValue(field, value);
+  payoutAmount.value = null;
+};
+
+const calculatePayoutAmount = async () => {
+  console.log(props.form.model.fromDate.value.value, props.form.model.toDate.value.value);
```
Comment (CodeNamedRobin) [inline apps/dashboard/src/modules/seller/components/seller/forms/SellerPayoutCreateForm.vue]: beep boop

File: apps/dashboard/src/modules/seller/components/seller/forms/SellerPayoutCreateForm.vue [hunk 20]
```diff
@@ -0,0 +1,167 @@
...
+  const amount = await ApiService.user.getUsersSalesReport(props.seller.id, props.form.model.fromDate.value.value, props.form.model.toDate.value.value);
+  console.error(amount.data);
```
Comment (CodeNamedRobin) [inline apps/dashboard/src/modules/seller/components/seller/forms/SellerPayoutCreateForm.vue]: beep boop

File: apps/dashboard/src/modules/seller/components/seller/forms/SellerPayoutCreateForm.vue [hunk 21]
```diff
@@ -0,0 +1,167 @@
...
+
+  } catch (error) {
+    console.error('Error downloading PDF:', error);
+    pdfLoading.value = false;
+  }
+};
+
+const emit = defineEmits(['update:edit']);
+setSubmit(props.form, props.form.context.handleSubmit(async (values) => {
+  // setSuccess(props.form, null);
```
Comment (CodeNamedRobin) [inline apps/dashboard/src/modules/seller/components/seller/forms/SellerPayoutCreateForm.vue]: Commented code

File: apps/dashboard/src/modules/seller/components/seller/forms/SellerPayoutCreateForm.vue [hunk 22]
```diff
@@ -0,0 +1,167 @@
...
+    return;
+  }
+  const request: CreateSellerPayoutRequest = {
+    endDate: values.toDate,
+    reference: values.reference,
+    requestedById: values.user.id,
+    startDate: values.fromDate,
+  };
+  await sellerPayoutStore.createPayout(request).then((res) => {
+    console.error(res);
```
Comment (CodeNamedRobin) [inline apps/dashboard/src/modules/seller/components/seller/forms/SellerPayoutCreateForm.vue]: beep bop

File: apps/dashboard/src/modules/seller/components/seller/forms/SellerPayoutCreateForm.vue [hunk 23]
```diff
@@ -0,0 +1,167 @@
...
+    setSuccess(props.form, res);
+  }).catch((err) => {
+    console.error(err);
+    setSuccess(props.form, null);
+  });
+  console.error(values);
```
Comment (CodeNamedRobin) [inline apps/dashboard/src/modules/seller/components/seller/forms/SellerPayoutCreateForm.vue]: boop beep

File: apps/dashboard/src/modules/seller/components/seller/forms/SellerPayoutCreateForm.vue [hunk 24]
```diff
@@ -0,0 +1,167 @@
...
+    endDate: values.toDate,
+    reference: values.reference,
+    requestedById: values.user.id,
+    startDate: values.fromDate,
+  };
+  await sellerPayoutStore.createPayout(request).then((res) => {
+    console.error(res);
+    setSuccess(props.form, res);
+  }).catch((err) => {
+    console.error(err);
```
Comment (CodeNamedRobin) [inline apps/dashboard/src/modules/seller/components/seller/forms/SellerPayoutCreateForm.vue]: beep beep

File: apps/dashboard/src/modules/seller/views/SellerPayoutsView.vue [hunk 25]
```diff
@@ -0,0 +1,62 @@
...
+const balance: Ref<BalanceResponse | null> = ref(null);
+const zeroBalance = computed(() => {
+  return balance.value?.amount?.amount === 0;
+});
+const user: Ref<UserResponse | undefined> = ref();
+
+const form = schemaToForm(createSellerPayoutObject);
+
+onBeforeMount(async () => {
+  console.error(route.params);
```
Comment (CodeNamedRobin) [inline apps/dashboard/src/modules/seller/views/SellerPayoutsView.vue]: beop beop

File: apps/dashboard/src/modules/seller/views/SellerPayoutsView.vue [hunk 26]
```diff
@@ -0,0 +1,62 @@
...
+  const id = Number(route.params.id);
+  await ApiService.user.getIndividualUser(id).then((res) => {
+    user.value = res.data;
+    form.context.resetForm({ values: { user: res.data } });
+  }).catch(async (e) => {
+    console.error(e);
```
Comment (CodeNamedRobin) [inline apps/dashboard/src/modules/seller/views/SellerPayoutsView.vue]: beep boop

File: apps/dashboard/src/modules/seller/views/SellerPayoutsView.vue [hunk 27]
```diff
@@ -0,0 +1,62 @@
...
+    // await router.replace('/error');
```
Comment (CodeNamedRobin) [inline apps/dashboard/src/modules/seller/views/SellerPayoutsView.vue]: commented code

File: apps/point-of-sale/src/components/Cart/TransactionHistory/TransactionHistoryRowComponent.vue [hunk 28]
```diff
@@ -11,7 +11,7 @@
     </div>
       <div v-if="settings.isBorrelmode">
         <div class="mt-1 fs-6">
-          Transaction for {{ transaction.from.firstName}} by {{transaction.createdBy?.firstName}}
+          Transaction for {{ transaction.from.firstName }} by {{ transaction.createdBy?.firstName }}
```
Comment (CodeNamedRobin) [inline apps/point-of-sale/src/components/Cart/TransactionHistory/TransactionHistoryRowComponent.vue:14]: Almost correctly translated.

**Review by CodeNamedRobin — CHANGES_REQUESTED** [review]
Mostly small things.

File: apps/dashboard/src/components/TopNavbar.vue (hunk 5 above)
Comment (JustSamuel) [reply apps/dashboard/src/components/TopNavbar.vue:128]: `''` is equal to no notification.

File: apps/point-of-sale/src/components/Cart/TransactionHistory/TransactionHistoryRowComponent.vue (hunk 28 above)
Comment (JustSamuel) [reply apps/point-of-sale/src/components/Cart/TransactionHistory/TransactionHistoryRowComponent.vue:14]: not my pr

File: apps/dashboard/src/main.ts (hunk 9 above)
Comment (JustSamuel) [reply apps/dashboard/src/main.ts:84]: its already there

File: apps/dashboard/src/locales/en/modules/seller.json (hunk 8 above)
Comment (JustSamuel) [reply apps/dashboard/src/locales/en/modules/seller.json:61]: seller payouts are all under the seller module. When we have RBAC access I will rewrite it such that users can _view_ their payouts.

**Review by CodeNamedRobin — APPROVED** [review]

**Pull request merged**